### PR TITLE
Add `ScheduledDeletion` State

### PR DESCRIPTION
**Pull request by joshuajerin**

### CODE REVIEW

**Comment by joshuajerin** [comment]
From testing in dev, we should be able to restore an instance without the coredbspec and namespace

File: conductor/src/main.rs [hunk 1]
```diff
@@ -549,61 +549,69 @@ async fn run(metrics: CustomMetrics) -> Result<(), ConductorError> {
                 info!("{}: Deleting namespace {}", read_msg.msg_id, &namespace);
                 delete_namespace(client.clone(), &namespace).await?;
 
-                if is_cloud_formation {
-                    info!("{}: Deleting cloudformation stack", read_msg.msg_id);
-                    delete_cloudformation(aws_region.clone(), &namespace).await?;
-                }
+                if read_msg.message.event_type == Event::Delete {
```
Comment (nhudson) [inline conductor/src/main.rs:552]: IMHO this if statement isn't needed.  If the instance is paused, we have already removed the IAM/Workload Idenitity/UAMI from the cloud provider anyway.  It's safe to run for both events (`Delete` and `ScheduledDeletion`).

**Review by nhudson — APPROVED** [review]

File: conductor/src/types.rs [hunk 2]
```diff
@@ -33,6 +33,8 @@ pub enum Event {
     Started,
     Restore,
     Restored,
+    ScheduleDeletion,
+    ScheduleDeletionComplete,
```
Comment (ChuckHend) [inline conductor/src/types.rs:37]: Make sure these align exactly with what the control-plane is expecting when serialized.

**Comment by joshuajerin** [comment]
We decided not to add any new states so I'm closing this ticket

**Pull request closed without merging**